src/controller/python/chip/clusters/Types.py

```diff
@@ -32,5 +32,7 @@ def __ne__(self, other):
     def __lt__(self, other):
         return True
 
+    def __hash__(self):
```

Comment:
Why is this the right fix as opposed to just using `default_factory` as the error message says?

Reply:
That works too, but would need to be added to each Nullable field in `Objects.py`. 

The idea of the chosen solution is mentioned in https://github.com/huggingface/datasets/issues/5230:

> 1. Changed in version 3.11: Instead of looking for and disallowing objects of type list, dict, or set, unhashable objects are now not allowed as default values. Unhashability is used to approximate mutability.

Making `Nullable` hashable makes the type eligible as direct default value.

Reading [parameters to dataclass](https://docs.python.org/3/library/dataclasses.html) a bit in more depth, probably using frozen makes even more sense. I'll give this a try.

Reply:
Just setting `frozen=True` using the dataclass decorator does resolve the issue as well.

However, with that dataclass will generate a `__hash__()` function, according to the documentation (eq is set to True by default):

> If eq and frozen are both true, by default [dataclass()](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) will generate a [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) method for you.

So, instead of relying on dataclass smarts, just implement our own is probably a good idea still. Especially since the type is so trivial.